apps/api/src/app/environments-v1/usecases/output-renderers/render-email-output.usecase.ts

```diff
@@ -15,16 +16,32 @@ export class RenderEmailOutputUsecase {
 
   @InstrumentUsecase()
   async execute(renderCommand: RenderEmailOutputCommand): Promise<EmailRenderOutput> {
-    const { body, subject } = EmailStepControlZodSchema.parse(renderCommand.controlValues);
+    const { body, subject } = emailStepControlZodSchema.parse(renderCommand.controlValues);
 
     if (isEmpty(body)) {
       return { subject, body: '' };
     }
 
-    const expandedSchema = this.transformForAndShowLogic(body, renderCommand.fullPayloadForRender);
-    const htmlRendered = await this.renderEmail(expandedSchema);
+    const expandedMailyContent = this.transformForAndShowLogic(body, renderCommand.fullPayloadForRender);
+    const parsedTipTap = await this.parseTipTapNodeByLiquid(expandedMailyContent, renderCommand);
+    const renderedHtml = await this.renderEmail(parsedTipTap);
 
-    return { subject, body: htmlRendered };
+    return { subject, body: renderedHtml };
+  }
+
+  private async parseTipTapNodeByLiquid(
+    value: TipTapNode,
+    renderCommand: RenderEmailOutputCommand
+  ): Promise<TipTapNode> {
+    const client = new Liquid();
+    const templateString = client.parse(JSON.stringify(value));
+    const parsedTipTap = await client.render(templateString, {
+      payload: renderCommand.fullPayloadForRender.payload,
```

Comment:
Can we get rid of the `fullPayloadForRender` extra nesting? It will make the code way more beautiful.

Reply:
I believe we can, but this refactor is focused on doing the minimum needed to make the code work with what we currently have. We'll try to avoid major changes as much as possible. 

The goal is to ensure the use cases—like sending a simple email, "show" component, then "for" components—work as expected with minimal refactoring. If that's not possible, we'll consider a bigger refactor.

Once we have a working solution, we can refactor further if needed. One potential improvement could be flattening the data as you suggested.